src/serialization/CCLACSets.jl

```diff
@@ -0,0 +1,180 @@
+module CCLACSets
+export read_ccl_acset
+
+using ...ACSetInterface, ...Schemas, ..ACSetSerialization, ...DenseACSets
+
+using ACSets
+
+using JSON3
+using MLStyle
+
+@data AnElement begin
+    ObjectElem(name::String, type::String)
+    HomElem(name::String, dom::String, cod::String)
+end
+
+struct CCLModel
+    name::Symbol
+    elems::Dict{String, AnElement}
+    function CCLModel(json::AbstractDict, predecessor=nothing)
+    dict = Dict{String,AnElement}()
+    foreach(json.notebook.cells) do cell
+        @match cell.content.tag begin
+            "object" => 
+            push!(dict, cell.content[:id] => ObjectElem(cell.content[:name], ""))
+            "morphism" => begin
+                name = 
+                if isempty(cell.content[:name])
+                    if isnothing(cell.content[:over])
+                        cell.content[:name]
+                    else
+                        cell.content[:over][:content]
+                    end
+                else
+                    cell.content[:name]
+                end
+                push!(dict, cell.content[:id] => 
+                HomElem(name, 
+                     cell.content[:dom][:content], cell.content[:cod][:content]))
+            end
+        end
+    end
+    new(Symbol(json.theory), dict)
+end 
+end
+export CCLModel
+
+struct CCLDiagram
```

Comment:
Many of my comments about `CCLModel` apply here as well.

More importantly, the logic handling the `over` field is not right. Each object (resp. morphism) in the diagram is over (maps to) an object (resp. morphism) in a model, and it's important to capture that.

Compare with the `DiagramObDecl` and `DiagramMorDecl` structs [here](https://github.com/ToposInstitute/CatColab/blob/main/packages/catlog-wasm/src/model_diagram.rs).